Non discovery methods decorator

jlee1-made opened the pull request:
Allow marking methods as NOT involved in test discovery
    
    i.e. not an examples method, not an assertion method, etc. etc.
    
    Example (with_context_manager is just a regular method and is ignored by
    contexts except to record that fact -- in particular it is not treated as a
    setup method despite the word "context" appearing in the method name):
    
        from contexts import method
    
        class When_whatever:
    
            @method
            def with_context_manager(self):
                ...
    
            def should_whatever(self):
                ...

jlee1-made committed:
Allow marking methods as NOT involved in test discovery

i.e. not an examples method, not an assertion method, etc. etc.

Example (with_context_manager is just a regular method and is ignored by
contexts except to record that fact -- in particular it is not treated as a
setup method despite the word "context" appearing in the method name):

    from contexts import method

    class When_whatever:

        @method
        def with_context_manager(self):
            ...

        def should_whatever(self):
            ...

jlee1-made committed:
Add contexts.method to __all__

hjwp commented:
may i bikeshed on the name?  how about `@contexts_ignore`?  more self-explanatory...

jlee1-made commented:
> may i bikeshed on the name? how about `@contexts_ignore`? more self-explanatory...

I don't like `method` and considered `ignore` and rejected it as not fitting the rest of the list of decorators that I do think it should fit: "this is some `@examples`, this is a `@setup', this is an `@action`, this is an `@assertion`, this is a `@teardown`, this is an... `@ignore`" :-/

So: how about `implementation`?

hjwp commented on test/plugin_tests/identification_tests/decorator_tests.py:
```diff
@@ -35,6 +35,23 @@ def it_should_identify_it_as_a_context(self):
+        self.result = self.identifier.identify_method(self.method)
+
+    @assertion
+    def it_should_identify_it_as_examples(self):
```
typo

jlee1-made commented:
We had a good long bikeshed and settled on `ignore`.  Harry didn't like `implementation` as being too opaque, I didn't like `normal_method` for being wtf is a normal method.

Any more for the naming discussion before I make that change?

jlee1-made commented:
Csaba suggests `ignored` or `excluded` (instead of `ignore`).  They're better at least in fitting better grammatically (if not in meaning).

I think maybe `excluded` is better than `ignored` because plugins will be able to mark methods as excluded, and "ignored" suggested that `Contexts` really doesn't "see" these methods at all, which isn't quite right, for that reason?

hjwp commented:
if i had to choose, i'd choose "ignored" (naturally), but i think both are pretty good.

jlee1-made commented:
Naturally?

hjwp commented:
like, naturally, why would i ever make it easy for myself by just agreeing with someone?

hjwp commented:
anyways, ignored or excluded, either sound fine, if you make the change i'd say let's go for it.  apparently we'lkl have to make travis happy first tho?

jlee1-made committed:
Rename `method` -> `ignored`

See bikeshedding on PR https://github.com/madedotcom/Contexts/pull/7

jlee1-made committed:
pyflakes is complaining about where I put my + signs

I never even understood this one (W504 line break after binary operator).

pyflakes is so useful, but I don't want it nagging at me from CI about coding style

jlee1-made committed:
Hm more pyflakes

jlee1-made committed:
More pyflakes/flake8

jlee1-made committed:
Consistency with rest of docs
